core/chain_makers.go

```diff
@@ -238,58 +243,79 @@ func GenerateChain(config *params.ChainConfig, parent *types.Block, engine conse
-	var time uint64
-	if parent.Time() == 0 {
-		time = 10
-	} else {
-		time = parent.Time() + 10 // block time is fixed at 10 seconds
+func makeHeader(config *params.ChainConfig, chain consensus.ChainReader, parent *types.Block, order int, state *state.StateDB, engine consensus.Engine) *types.Header {
+	// return genesis block as itself for first block in chain
+	if parent.Header().Number[0].Cmp(big.NewInt(0)) == 0 {
+		return parent.Header()
 	}
 
+	// initialization of header
 	baseFee := misc.CalcBaseFee(chain.Config(), parent.Header(), chain.GetHeaderByNumber, chain.GetUnclesInChain, chain.GetGasUsedInChain)
-
 	header := &types.Header{
-		Coinbase:    []common.Address{common.Address{}, common.Address{}, common.Address{}},
-		Number:      []*big.Int{big.NewInt(int64(1)), big.NewInt(int64(1)), big.NewInt(int64(1))},
-		ParentHash:  []common.Hash{common.Hash{}, common.Hash{}, common.Hash{}},
-		Root:        []common.Hash{common.Hash{}, common.Hash{}, common.Hash{}},
-		Difficulty:  []*big.Int{big.NewInt(1), big.NewInt(1), big.NewInt(1)},
-		UncleHash:   types.EmptyUncleHash,
-		TxHash:      types.EmptyRootHash,
-		ReceiptHash: types.EmptyRootHash,
-		Bloom:       []types.Bloom{types.Bloom{}, types.Bloom{}, types.Bloom{}},
-		Time:        time,
-		BaseFee:     []*big.Int{baseFee, baseFee, baseFee},
-		GasLimit:    []uint64{0, 0, 0},
-		GasUsed:     []uint64{0, 0, 0},
-		Extra:       [][]byte{[]byte(nil), []byte(nil), []byte(nil)},
+		ParentHash:        make([]common.Hash, 3),
+		Number:            []*big.Int{new(big.Int).SetUint64(0), new(big.Int).SetUint64(0), new(big.Int).SetUint64(0)},
+		Extra:             make([][]byte, 3),
+		Time:              uint64(0),
+		BaseFee:           []*big.Int{baseFee, baseFee, baseFee},
+		GasLimit:          make([]uint64, 3),
+		Coinbase:          make([]common.Address, 3),
+		Difficulty:        make([]*big.Int, 3),
+		NetworkDifficulty: make([]*big.Int, 3),
+		Root:              make([]common.Hash, 3),
+		TxHash:            make([]common.Hash, 3),
+		ReceiptHash:       make([]common.Hash, 3),
+		GasUsed:           make([]uint64, 3),
+		Bloom:             make([]types.Bloom, 3),
+		Location:          chain.Config().Location,
 	}
-	header.GasLimit[types.QuaiNetworkContext] = parent.GasLimit()
 
-	parentHeader := &types.Header{
-		Number:     []*big.Int{big.NewInt(int64(1)), big.NewInt(int64(1)), big.NewInt(int64(1))},
-		Difficulty: []*big.Int{big.NewInt(1), big.NewInt(1), big.NewInt(1)},
-		UncleHash:  types.EmptyUncleHash,
-		Time:       time - 10,
-	}
-
-	if chain.Config().IsLondon(header.Number[types.QuaiNetworkContext]) {
-		if !chain.Config().IsLondon(parent.Number()) {
-			parentGasLimit := parent.GasLimit() * params.ElasticityMultiplier
-			header.GasLimit[types.QuaiNetworkContext] = CalcGasLimit(parentGasLimit, parent.GasUsed(), 0)
-		}
+	// same across orders
+	header.Time = parent.Time() + uint64(10)
```

Comment:
I think this is fine (pretty sure the fakepow mode will ignore this), but leaving this note as a reminder to come back and think this through. There may be cases where  we need to set times more meaningfully, especially for the V2 generator which may be generating blocks on different chains iteratively.

Reply:
Yes, this may need some change in the future depending on how/if we want time handled differently during testing. For certain tests we may want to manually assign times for blocks. For the most basic testing case, it is only important that the time is consistent for resulting hashes/parenthashes.